src/main/java/org/skriptlang/skript/misc/colours/ColourModule.java

```diff
@@ -0,0 +1,110 @@
+package org.skriptlang.skript.misc.colours;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.skript.lang.function.Parameter;
+import ch.njol.skript.lang.function.SimpleJavaFunction;
+import ch.njol.skript.lang.util.SimpleLiteral;
+import ch.njol.skript.registrations.DefaultClasses;
+import ch.njol.skript.util.Color;
+import ch.njol.skript.util.ColorRGB;
+import ch.njol.util.coll.CollectionUtils;
+
+import java.io.IOException;
+
+public class ColourModule {
+
+	public static void load() throws IOException {
+		Skript.getAddonInstance().loadClasses("org.skriptlang.skript.misc", "colours");
+
+		Functions.registerFunction(new SimpleJavaFunction<Color>("shade", new Parameter[] {
+			new Parameter<>("colour", DefaultClasses.COLOR, true, null),
+			new Parameter<>("amount", DefaultClasses.LONG, true, new SimpleLiteral<>(1L, true)),
+			new Parameter<>("hsl", DefaultClasses.BOOLEAN, true, new SimpleLiteral<>(false, true))
+		}, DefaultClasses.COLOR, true) {
+			@Override
+			public ColorRGB[] executeSimple(Object[][] params) {
+				Color colour = (Color) params[0][0];
+				Long amount = (Long) params[1][0];
+				boolean hsl = (Boolean) params[2][0];
+				return CollectionUtils.array(hsl
+					? ColourUtils.shadeColorHSL(colour, amount.intValue())
+					: ColourUtils.shadeColor(colour, amount.intValue()));
+			}
+		}).description(
+			"Shades a given colour by a given amount. Optionally can use HSL methods to achieve this instead.",
+			"The amount should be from 0-100, with 0 doing nothing, and 100 shading it all the way to black."
+		).examples(
+			"set {_darkRed} to shade(red, 10)",
+			"set {_darkerRed} to shade(red, 20)"
+		).since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Color>("tint", new Parameter[] {
+			new Parameter<>("colour", DefaultClasses.COLOR, true, null),
+			new Parameter<>("amount", DefaultClasses.LONG, true, new SimpleLiteral<>(1L, true)),
+			new Parameter<>("hsl", DefaultClasses.BOOLEAN, true, new SimpleLiteral<>(false, true))
+		}, DefaultClasses.COLOR, true) {
+			@Override
+			public ColorRGB[] executeSimple(Object[][] params) {
+				Color colour = (Color) params[0][0];
+				Long amount = (Long) params[1][0];
+				boolean hsl = (Boolean) params[2][0];
+				return CollectionUtils.array(hsl
+					? ColourUtils.tintColorHSL(colour, amount.intValue())
+					: ColourUtils.tintColor(colour, amount.intValue()));
+			}
+		}).description(
+			"Tints a given colour by a given amount. Optionally can use HSL methods to achieve this instead.",
+			"The amount should be from 0-100, with 0 doing nothing, and 100 tinting it all the way to white."
```

Comment:
the amount description implies it, but for this and shade it might be nice to explain exactly what they do more clearly